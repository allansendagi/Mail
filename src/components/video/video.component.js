import React from 'react';
import './video.styles.scss';

const BASE_EMBED_URL = 'https://www.youtube.com/embed/';

export function Video(props) {
	if (!props.id) {
		return null;
	}
	const embedUrl = `${BASE_EMBED_URL}${props.id}`;
	// const embedUrl = `${BASE_EMBED_URL}${props.id}?autoplay=1`;

	return (
		<div className='video-container'>
		 <div className='video'>
			<iframe className='video-player' src={embedUrl} frameborder="0"
			allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen title={'video'}></iframe>
		 </div>
		</div>
	);
}