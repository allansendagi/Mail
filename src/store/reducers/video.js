import {MOST_POPULAR, MOST_POPULAR_BY_CATEGORY, VIDEO_CATEGORIES} from '../actions/video';
import {SUCCESS} from '../actions';
import {createSelector} from 'reselect';


const initialState = {
  byId: {},
  mostPopular: {},
  categories: {},
  byCategory: {},
  related: {},


};

function reduceFetchMostPopularVideos(response, prevState) {
  const videoMap = response.items.reduce((accumulator, video) => {
    accumulator[video.id] = video;
    return accumulator;
  }, {});

  let items = Object.keys(videoMap);
  if (response.hasOwnProperty('prevPageToken') && prevState.mostPopular) {
    items = [...prevState.mostPopular.items, ...items];
  }

  const mostPopular = {
    totalResults: response.pageInfo.totalResults,
    nextPageToken: response.nextPageToken,
    items,
  };

  return {
    ...prevState,
    mostPopular,
    byId: {...prevState.byId, ...videoMap},
  };
}

export const getVideoCategoryIds = createSelector(
  state => state.videos.categories,
  (categories) => {
    return Object.keys(categories || {});
  }
);

export default function videos(state = initialState, action) {
  switch (action.type) {
    case MOST_POPULAR[SUCCESS]:
      return reduceFetchMostPopularVideos(action.response, state);
      case VIDEO_CATEGORIES[SUCCESS]:
      return reduceFetchVideoCategories(action.response, state);
      case MOST_POPULAR_BY_CATEGORY[SUCCESS]:
      return reduceFetchMostPopularVideosByCategory(action.response, action.categories, state);
    default:
      return state;
  }
}

export const getMostPopularVideos = createSelector(
  (state) => state.videos.byId,
  (state) => state.videos.mostPopular,
  (videosById, mostPopular) => {
    if (!mostPopular || !mostPopular.items) {
      return [];
    }
    return mostPopular.items.map(videoId => videosById[videoId]);
  }
);
function reduceFetchVideoCategories(response, prevState) {
  const categoryMapping = response.items.reduce((accumulator, category) => {
    accumulator[category.id] = category.snippet.title;
    return accumulator;
  }, {});
  return {
    ...prevState,
    categories: categoryMapping,
  };
}
function reduceFetchMostPopularVideosByCategory(responses, categories, prevState) {
  let videoMap = {};
  let byCategoryMap = {};

  responses.forEach((response, index) => {
    // ignore answer if there was an error
    if (response.status === 400) return;

    const categoryId = categories[index];
    const {byId, byCategory} = groupVideosByIdAndCategory(response.result);
    videoMap = {...videoMap, ...byId};
    byCategoryMap[categoryId] = byCategory;
  });

  // compute new state
  return {
    ...prevState,
    byId: {...prevState.byId, ...videoMap},
    byCategory: {...prevState.byCategory, ...byCategoryMap},
  };
}






