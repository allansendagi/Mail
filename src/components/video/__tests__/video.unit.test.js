import React from 'react';
import {Video} from '../video.component';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


configure({adapter: new Adapter()});
test('renders video component correctly', () => {
  const wrapper = shallow(
    <Video id='HAuXJVI_bUs'/>
  );
  expect(wrapper).toMatchSnapshot();
});

test('renders null if id in video component not specified', () => {
  const wrapper = shallow(
    <Video/>
  );
  expect(wrapper).toMatchSnapshot();
});