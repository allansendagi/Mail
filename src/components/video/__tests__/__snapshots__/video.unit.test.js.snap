// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders null if id in video component not specified 1`] = `ShallowWrapper {}`;

exports[`renders video component correctly 1`] = `ShallowWrapper {}`;
